src/foam/nanos/auth/PreventDuplicateEmailLoginService.js

```diff
@@ -26,19 +28,23 @@ foam.CLASS({
     {
       name: 'getUser',
       javaCode: `
-        DAO userDAO = (DAO) x.get("localUserUserDAO");
-        Sink sink = new ArraySink();
-        sink = userDAO
-          .where(OR(
+        DAO userDAO = ((DAO) x.get("localUserUserDAO")).where(
+          OR(
             EQ(User.EMAIL, identifier.toLowerCase()),
-            EQ(User.USER_NAME, identifier)))
-          .limit(2).select(sink);
+            EQ(User.USER_NAME, identifier)));
 
-        List list = ((ArraySink) sink).getArray();
-        if ( list.size() != 1 ) {
-          return null;
-        }
-        return (User) list.get(0);
+        // try to find a user under the theme spid
+        List users = ((ArraySink) userDAO
```

Comment:
This could be
Count count = (Count) userDAO.where(..).select(new Count());
then test count.getValue() 
This will use the count on the mdao index.